### PR TITLE
Maintenance mode

### PR DESCRIPTION
**Pull request by cdmo**
#291 

Will add the maintenance mode check to the patrons controller too after that PR is merged. Will add specs too. Just posting progress so approach can be seen.

### CODE REVIEW

File: spec/views/layouts/application.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'layouts/application', type: :view do
+  let(:mock_patron) { instance_double Patron, id: 'idhere' }
+
+  before do
+    without_partial_double_verification do
```
Comment (cdmo) [inline spec/views/layouts/application.html.erb_spec.rb:9]: this came from here: https://github.com/rspec/rspec-rails/issues/780#issuecomment-293111014

without it I couldn't mock this

I tried mocking ApplicationController#patron but rspec claimed that ApplicationController didn't implement `patron` 🤷

**Comment by banukutlu** [comment]
Screen Shot 2020-10-26 at 11 56 16 AM

Should we hide cancel button as well?

**Comment by cdmo** [comment]
oh forgot that little guy, yeah we should

File: spec/components/select_all_child_component_spec.rb [hunk 2]
```diff
@@ -7,6 +7,18 @@
 
   let(:component) { render_inline(described_class.new(select_all_child_args)).to_html }
 
+  context 'when site is in maintenance mode' do
+    after do
+      Settings.maintenance_mode = false
+    end
+
+    it 'makes a child checkbox with that name' do
```
Comment (banukutlu) [inline spec/components/select_all_child_component_spec.rb:15]: I did not understand this description 🤔

Comment (cdmo) [reply spec/components/select_all_child_component_spec.rb]: copy/paste mistake

**Comment by banukutlu** [comment]
> Screen Shot 2020-10-26 at 11 56 16 AM
> 
> @cdmo, should we hide cancel button as well?

also the Select column heading?

File: spec/components/select_all_component_spec.rb [hunk 3]
```diff
@@ -7,6 +7,18 @@
 
   let(:component) { render_inline(described_class.new(select_all_args)).to_html }
 
+  context 'when site is in maintenance mode' do
+    after do
+      Settings.maintenance_mode = false
+    end
+
+    it 'makes a child checkbox with that name' do
```
Comment (banukutlu) [inline spec/components/select_all_component_spec.rb:15]:
```suggestion
     it 'does not render' do
```

**Comment by cdmo** [comment]
oh you think so? Sure. I'll work on that.

File: spec/features/holds_spec.rb [hunk 4]
```diff
@@ -120,4 +120,20 @@
       expect(page).to have_css '#hold3911148'
     end
   end
+
+  context 'when site is in maintenance mode' do
+    after do
+      Settings.maintenance_mode = false
+    end
+
+    it 'makes a child checkbox with that name', js: true do
```
Comment (banukutlu) [inline spec/features/holds_spec.rb:129]: description needs to be updated here as well?

Comment (cdmo) [reply spec/features/holds_spec.rb:129]: yup

File: spec/views/layouts/application.html.erb_spec.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      allow(view).to receive(:patron).and_return(mock_patron)
+    end
+  end
+
+  context 'when site is in not in maintenance mode' do
+    after do
+      Settings.maintenance_mode = false
+    end
+
+    it 'makes a child checkbox with that name' do
```
Comment (banukutlu) [inline spec/views/layouts/application.html.erb_spec.rb:19]: description needs to be updated here as well

File: spec/views/layouts/application.html.erb_spec.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      expect(rendered).not_to have_text 'System Maintenance'
+    end
+  end
+
+  context 'when site is in maintenance mode' do
+    after do
+      Settings.maintenance_mode = false
+    end
+
+    it 'makes a child checkbox with that name' do
```
Comment (banukutlu) [inline spec/views/layouts/application.html.erb_spec.rb:33]: and here

File: spec/views/layouts/application.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+RSpec.describe 'layouts/application', type: :view do
+  let(:mock_patron) { instance_double Patron, id: 'idhere' }
+
+  before do
+    without_partial_double_verification do
+      allow(view).to receive(:patron).and_return(mock_patron)
+    end
+  end
+
+  context 'when site is in not in maintenance mode' do
```
Comment (banukutlu) [inline spec/views/layouts/application.html.erb_spec.rb:14]:
```suggestion
  context 'when site is in maintenance mode' do
```

**Comment by banukutlu** [comment]
> oh you think so? Sure. I'll work on that.

the other table does not have it also the renews so I thought it would be more consistent 🤔

**Review by banukutlu — CHANGES_REQUESTED** [review]
overall looks great, just couple minor requests in the comments

File: spec/views/layouts/application.html.erb_spec.rb (hunk 6 above)
Comment (cdmo) [reply spec/views/layouts/application.html.erb_spec.rb:33]: 🤦

File: spec/views/layouts/application.html.erb_spec.rb (hunk 5 above)
Comment (cdmo) [reply spec/views/layouts/application.html.erb_spec.rb:19]: 🤭

File: spec/views/layouts/application.html.erb_spec.rb (hunk 7 above)
Comment (cdmo) [reply spec/views/layouts/application.html.erb_spec.rb:14]: 🥵

**Comment by cdmo** [comment]
The "Holds that are not ready for pickup" table _never_ has a "Select" label.

**Comment by banukutlu** [comment]
> The "Holds that are not ready for pickup" table _never_ has a "Select" label.

checkouts table does not have it as well, wonder why we kept Select label for this one then 🤔

**Comment by cdmo** [comment]
Just a detail we missed I guess? What do you think we should do then?

**Comment by banukutlu** [comment]
> Just a detail we missed I guess? What do you think we should do then?

I am inclined to just get rid of `Select` label, not sure if it adds any value 🤔  what do you think?

**Comment by cdmo** [comment]
works for me, I'll change it here

**Comment by banukutlu** [comment]
> works for me, I'll change it here

thanks!

File: spec/features/holds_spec.rb (hunk 4 above)
Comment (banukutlu) [reply spec/features/holds_spec.rb:129]: only this description left to update ☝️

**Review by banukutlu — APPROVED** [review]
thanks 👍

**Pull request merged**